test/numerical/Common.hpp

```diff
@@ -3,14 +3,18 @@
  */
 
 #if defined(_MSC_VER)
+#ifndef _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS
```

Comment:
Since we are now saying that this might come from some place other than us, please save the save of the macro and then restore it below - don't just `undef` always since it may have been set by someone other than us.

Reply:
Something like:

```
#ifndef _SILENCE
#define _SILENCE
#else 
#define _SILENCE_WAS_DEFINED
```

```
#ifdef _SILENCE && !defined SILENCE_WAS_DEFINED
#undef _SILENCE
```

Reply:
I would say something like: `if there's a dependency on ORT *or another third party library*...`